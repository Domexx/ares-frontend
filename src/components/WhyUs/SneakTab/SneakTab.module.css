/* === === === === WhyUs/SneakTab === === === === */

.SneakTab {
  margin-top: 40px;
}

.SneakTab-item {
  display: flex;
}

.SneakTab-itemContent {
  z-index: 1;
  display: flex;
  flex-direction: column;
  align-self: center;
}

.SneakTab-itemDescription {
  font-size: 16px;
  color: var(--clr-dark-200);
}

.SneakTab-item:nth-of-type(1) {
  grid-row: 1;
  grid-column: 1 / span 7;
}

.SneakTab-item:nth-of-type(2) {
  grid-row: 2;
  grid-column: 7 / span 7;
}

.SneakTab-item:nth-of-type(3) {
  grid-row: 3;
  grid-column: 1 / span 7;
}

.SneakTab-item:nth-of-type(4) {
  grid-row: 4;
  grid-column: 7 / span 7;
}

.SneakTab-item:nth-of-type(5) {
  grid-row: 5;
  grid-column: 1 / span 7;
}

.SneakTab-item:nth-of-type(6) {
  grid-row: 6;
  grid-column: 7 / span 7;
  gap: 20px;
}

@media (--tablet-landscape) {
  .SneakTab-item {
    margin-top: 60px;
  }

  .SneakTab-item:nth-of-type(1) {
    grid-row: 1;
    grid-column: 2 / span 12;
  }

  .SneakTab-item:nth-of-type(2) {
    grid-row: 2;
    grid-column: 6 / span 10;
  }

  .SneakTab-item:nth-of-type(3) {
    grid-row: 3;
    grid-column: 2 / span 12;
  }

  .SneakTab-item:nth-of-type(4) {
    grid-row: 4;
    grid-column: 6 / span 10;
  }

  .SneakTab-item:nth-of-type(5) {
    grid-row: 5;
    grid-column: 2 / span 12;
  }

  .SneakTab-item:nth-of-type(6) {
    grid-row: 6;
    grid-column: 6 / span 10;
    gap: 20px;
  }
}

@media (--phone) {
  .SneakTab-item {
    flex-direction: column;
    grid-column: 1 / -1;
    gap: 30px;
  }

  .SneakTab-itemTitle {
    font-size: 20px;
  }

  .SneakTab-itemDescription {
    font-size: 12px;
  }

  .SneakTab-itemContent {
    padding: 0 30px 0 30px;
  }

  .SneakTab-item:nth-of-type(1) {
    grid-row: 1;
    grid-column: 1 / -1;
  }

  .SneakTab-item:nth-of-type(2) {
    grid-row: 2;
    grid-column: 1 / -1;
  }

  .SneakTab-item:nth-of-type(3) {
    grid-row: 3;
    grid-column: 1 / -1;
  }

  .SneakTab-item:nth-of-type(4) {
    grid-row: 4;
    grid-column: 1 / -1;
  }

  .SneakTab-item:nth-of-type(5) {
    grid-row: 5;
    grid-column: 1 / -1;
  }

  .SneakTab-item:nth-of-type(6) {
    grid-row: 6;
    grid-column: 1 / -1;
  }
}
