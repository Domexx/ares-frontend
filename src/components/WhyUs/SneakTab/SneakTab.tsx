import React from 'react';
import Image from 'next/image';
import { Sneaks, SneaksItem } from '../../../shared/interfaces/WhyUsItem';
import s from './SneakTab.module.css';

export type Props = {
  main: Sneaks;
};

const SneakTab: React.FC<Props> = ({ main }) => {
  const items = main.items.map((item: SneaksItem, index: number) => {
    return (
      // eslint-disable-next-line react/no-array-index-key
      <div key={index} className={s['SneakTab-item']}>
        <Image layout="intrinsic" height="600" width="600" objectFit="contain" src={item.image} alt={item.title} />
        <div className={s['SneakTab-itemContent']}>
          <div className={`${s['SneakTab-itemTitle']} title`}>{item.title}</div>
          <p className={s['SneakTab-itemDescription']}>{item.description}</p>
        </div>
      </div>
    );
  });
  return <div className={`${s['SneakTab']} u-grid`}>{items}</div>;
};

export default SneakTab;
