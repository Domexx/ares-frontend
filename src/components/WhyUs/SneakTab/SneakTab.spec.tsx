import React from 'react';

import { render } from '@testing-library/react';

import SneakTab from './SneakTab';

describe('Sneak Tab Component', () => {
  it('should render properly', () => {
    const { container } = render(
      <SneakTab
        main={{
          title: 'Sneaks',
          items: [
            {
              title: 'Profile Completion',
              description:
                'In Habbo.so you have the possibility that we support you with your profile creation. We will tell you exactly what you need to do and when you can finally show your perfect profile to the world.',
              image: '/static/images/Maintenance/Profile.png',
            },
            {
              title: 'Profile Completion',
              description:
                'In Habbo.so you have the possibility that we support you with your profile creation. We will tell you exactly what you need to do and when you can finally show your perfect profile to the world.',
              image: '/static/images/Maintenance/Groups.png',
            },
            {
              title: 'Profile Completion',
              description:
                'In Habbo.so you have the possibility that we support you with your profile creation. We will tell you exactly what you need to do and when you can finally show your perfect profile to the world.',
              image: '/static/images/Maintenance/Calendar.png',
            },
            {
              title: 'Profile Completion',
              description:
                'In Habbo.so you have the possibility that we support you with your profile creation. We will tell you exactly what you need to do and when you can finally show your perfect profile to the world.',
              image: '/static/images/Maintenance/Sidebar.png',
            },
            {
              title: 'Profile Completion',
              description:
                'In Habbo.so you have the possibility that we support you with your profile creation. We will tell you exactly what you need to do and when you can finally show your perfect profile to the world.',
              image: '/static/images/Maintenance/Navigator.png',
            },
            {
              title: 'Profile Completion',
              description:
                'In Habbo.so you have the possibility that we support you with your profile creation. We will tell you exactly what you need to do and when you can finally show your perfect profile to the world.',
              image: '/static/images/Maintenance/Chat.png',
            },
          ],
        }}
      />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
