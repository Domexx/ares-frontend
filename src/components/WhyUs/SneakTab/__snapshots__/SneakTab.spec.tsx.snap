// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sneak Tab Component should render properly 1`] = `
<div
  class="SneakTab u-grid"
>
  <div
    class="SneakTab-item"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27600%27%20height=%27600%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="Profile Completion"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
      />
      <noscript />
    </span>
    <div
      class="SneakTab-itemContent"
    >
      <div
        class="SneakTab-itemTitle title"
      >
        Profile Completion
      </div>
      <p
        class="SneakTab-itemDescription"
      >
        In Habbo.so you have the possibility that we support you with your profile creation. We will tell you exactly what you need to do and when you can finally show your perfect profile to the world.
      </p>
    </div>
  </div>
  <div
    class="SneakTab-item"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27600%27%20height=%27600%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="Profile Completion"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
      />
      <noscript />
    </span>
    <div
      class="SneakTab-itemContent"
    >
      <div
        class="SneakTab-itemTitle title"
      >
        Profile Completion
      </div>
      <p
        class="SneakTab-itemDescription"
      >
        In Habbo.so you have the possibility that we support you with your profile creation. We will tell you exactly what you need to do and when you can finally show your perfect profile to the world.
      </p>
    </div>
  </div>
  <div
    class="SneakTab-item"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27600%27%20height=%27600%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="Profile Completion"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
      />
      <noscript />
    </span>
    <div
      class="SneakTab-itemContent"
    >
      <div
        class="SneakTab-itemTitle title"
      >
        Profile Completion
      </div>
      <p
        class="SneakTab-itemDescription"
      >
        In Habbo.so you have the possibility that we support you with your profile creation. We will tell you exactly what you need to do and when you can finally show your perfect profile to the world.
      </p>
    </div>
  </div>
  <div
    class="SneakTab-item"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27600%27%20height=%27600%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="Profile Completion"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
      />
      <noscript />
    </span>
    <div
      class="SneakTab-itemContent"
    >
      <div
        class="SneakTab-itemTitle title"
      >
        Profile Completion
      </div>
      <p
        class="SneakTab-itemDescription"
      >
        In Habbo.so you have the possibility that we support you with your profile creation. We will tell you exactly what you need to do and when you can finally show your perfect profile to the world.
      </p>
    </div>
  </div>
  <div
    class="SneakTab-item"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27600%27%20height=%27600%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="Profile Completion"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
      />
      <noscript />
    </span>
    <div
      class="SneakTab-itemContent"
    >
      <div
        class="SneakTab-itemTitle title"
      >
        Profile Completion
      </div>
      <p
        class="SneakTab-itemDescription"
      >
        In Habbo.so you have the possibility that we support you with your profile creation. We will tell you exactly what you need to do and when you can finally show your perfect profile to the world.
      </p>
    </div>
  </div>
  <div
    class="SneakTab-item"
  >
    <span
      style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
      >
        <img
          alt=""
          aria-hidden="true"
          src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27600%27%20height=%27600%27/%3e"
          style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
        />
      </span>
      <img
        alt="Profile Completion"
        data-nimg="intrinsic"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
      />
      <noscript />
    </span>
    <div
      class="SneakTab-itemContent"
    >
      <div
        class="SneakTab-itemTitle title"
      >
        Profile Completion
      </div>
      <p
        class="SneakTab-itemDescription"
      >
        In Habbo.so you have the possibility that we support you with your profile creation. We will tell you exactly what you need to do and when you can finally show your perfect profile to the world.
      </p>
    </div>
  </div>
</div>
`;
